gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -965,7 +1021,7 @@ public <T> T fromJson(Reader json, Type typeOfT) throws JsonIOException, JsonSyn
   private static void assertFullConsumption(Object obj, JsonReader reader) {
     try {
       if (obj != null && reader.peek() != JsonToken.END_DOCUMENT) {
-        throw new JsonIOException("JSON document was not fully consumed.");
+        throw new JsonSyntaxException("JSON document was not fully consumed.");
```

Comment:
This fixes #1720

Reply:
This is a behaviour change, if minor, so I'll want to run it past all of Google's tests first.

Reply:
OK, no issues found.